packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -43,19 +79,38 @@ const createHistoryService = ({ strapi }: { strapi: LoadedStrapi }) => {
        * TODO: Fix the types for the middleware
        */
       strapi.documents.use(async (context, next) => {
-        // @ts-expect-error ContentType is not typed correctly on the context
-        const contentTypeUid = context.contentType.uid;
-        const params = context.args.at(-1) as any;
         // Ignore actions that don't mutate documents
-        if (!['create', 'update', 'publish', 'unpublish'].includes(context.action)) {
+        if (
+          !['create', 'update', 'publish', 'unpublish', 'discardDraft'].includes(context.action)
+        ) {
           return next(context);
         }
-
+        // @ts-expect-error ContentType is not typed correctly on the context
+        const contentTypeUid = context.contentType.uid;
         // Ignore content types not created by the user
         if (!contentTypeUid.startsWith('api::')) {
           return next(context);
         }
 
+        const result = (await next(context)) as any;
+        /**
+         * Publish and discardDraft actions store the created data on the result's versions array.
+         * Otherwise, we assume the data is the result.
+         *
+         * TODO: Fix unpublish which is broken since the data that was unpublished is not passed to the middleware
+         */
+        const data: NextDocument =
+          context.action === 'publish' || context.action === 'discardDraft'
+            ? result.versions[0]
+            : result;
+        /**
+         * The documentId should exist on the created data, fallback to context to handle the broken unpublish action
+         *
+         * TODO: Always get the documentId from the same place (data) once unpublish is fixed
+         */
+        const relatedDocumentId = data.documentId ?? context.args[0];
+        // Compute the status of the version
+        const status = await getVersionStatus(contentTypeUid, data);
         const fieldsToIgnore = [
```

Comment:
not sure of which status it is at the moment, but I believe localizations fields was (or is going to be )  removed

Reply:
ok sounds good thanks, I'll go ahead and remove it from the list here then